components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/token/JWTTokenIssuer.java

```diff
@@ -725,4 +730,27 @@ private JWTClaimsSet handleTokenBinding(JWTClaimsSet.Builder jwtClaimsSetBuilder
         }
         return jwtClaimsSetBuilder.build();
     }
+
+    /**
+     * Set signed tenant domain to the JWT token's realm claim.
+     * @param tenantDomain
+     * @param jwtClaimsSet
+     * @return
+     */
+    private JWTClaimsSet setSignerRealm(String tenantDomain, JWTClaimsSet jwtClaimsSet) {
+
+        Map<String, String> realm = new HashMap<>();
+        realm.put(OAuthConstants.OIDCClaims.TENANT, tenantDomain);
+        if (realm.size() > 0) {
```

Comment:
Won't the realm size always be > 0?

Reply:
Yes. It will be. As this check is unnecessary, I will remove.

Reply:
Re-added this check as the requirements changed.